main.tf

```diff
@@ -24,7 +24,7 @@ locals {
   role_arn = try(aws_iam_role.this[0].arn, var.iam_role_arn)
 
   create_outposts_local_cluster = var.outpost_config != null
-  enable_encryption_config      = length(var.encryption_config) > 0 && !local.create_outposts_local_cluster
```

Comment:
do you have a reproduction of what you are trying to do? I'm not quite following, and this logic change looks incorrect